sdk/Pulumi/Testing/IMocks.cs

```diff
@@ -78,4 +84,19 @@ public class MockCallArgs
         /// </summary>
         public string? Provider { get; set; }
     }
+
+    /// <summary>
+    /// MockRegisterResourceOutputsRequest for use in RegisterOutputRequest
+    /// </summary>
+    public class MockRegisterResourceOutputsRequest
+    {
+        /// <summary>
+        /// The URN of the resource of which the outputs are being registered
+        /// </summary>
+        public string? Urn { get; set; }
+        /// <summary>
+        /// The outputs which have been registered by the resource
+        /// </summary>
+        public ImmutableDictionary<string, object> Outputs { get; set; } = null!;
```

Comment:
Why ` = null!`?

Reply:
Was following the example above it but I changed the class into a record